src/macro-expansion.md

```diff
@@ -132,8 +132,8 @@ algorithm is exactly the same, but when the macro parser comes to a place in the
 current matcher where it needs to match a _non-terminal_ (e.g. `$mvar:ident`),
 it calls back to the normal Rust parser to get the contents of that
 non-terminal. In this case, the Rust parser would look for an `ident` token,
-which it finds (`foo`) and returns to the macro parser. Then, the macro parser
-proceeds in parsing as normal. Also, note that exactly one of the matchers from
+which it finds (`foo`), and returns to the macro parser. Then, the macro parser
+proceeds in parsing as normal. Also, note that only one of the matchers from
```

Comment:
I think `exactly` is more precise. `Only` could be taken as 0 or 1.

Reply:
The problem is with the "otherwise" clause here. Indeed, "exactly" is more precise", but it doesn't make sense when you have the "otherwise", since the "otherwise" clause is not predicated on there being zero matches.

Reply:
Ah, I see.

How about changing it to: if there is more than one match, the parse is ambiguous. If there are no matches at all, there is a syntax error.

Reply:
Sounds good!